src/SlingshotView.vala

```diff
@@ -287,27 +287,26 @@ public class Slingshot.SlingshotView : Gtk.Grid {
             case "7":
             case "8":
             case "9":
-                int page = int.parse (key);
-
-                if (event.state != Gdk.ModifierType.MOD1_MASK)
-                    return Gdk.EVENT_PROPAGATE;
-
-                if (modality == Modality.NORMAL_VIEW) {
-                    if (page < 0 || page == 9)
-                        grid_view.go_to_last ();
-                    else
-                        grid_view.go_to_number (page);
-                } else if (modality == Modality.CATEGORY_VIEW) {
-                    if (page < 0 || page == 9)
-                        category_view.app_view.go_to_last ();
-                    else
-                        category_view.app_view.go_to_number (page);
-                } else {
-                    return Gdk.EVENT_PROPAGATE;
+                if (event.state == Gdk.ModifierType.MOD1_MASK) {
+                    int page = int.parse (key);
+                    if (modality == Modality.NORMAL_VIEW) {
+                        if (page < 0 || page == 9) {
+                            grid_view.go_to_last ();
+                        } else {
+                            grid_view.go_to_number (page);
+                        }
+                    } else if (modality == Modality.CATEGORY_VIEW) {
+                        if (page < 0 || page == 9) {
+                            category_view.app_view.go_to_last ();
+                        } else {
+                            category_view.app_view.go_to_number (page);
+                        }
+                    }
```

Comment:
I guess we might want to break line after closing brackets if there are no another closing bracket or an else statement: https://elementary.io/docs/code/reference#whitespace